src/ts_catalog/continuous_agg.c

```diff
@@ -1487,34 +1499,137 @@ ts_continuous_agg_bucket_width(const ContinuousAgg *agg)
 }
 
 /*
- * Determines bucket size in months for given bucketing function.
+ * Calls one of time_bukcket_ng() versions depending on the arguments. This is
+ * a common procedure used by ts_compute_* below.
+ */
+static Datum
+generic_time_bucket_ng(const ContinuousAggsBucketFunction *bf, Datum timestamp)
+{
+	if (strlen(bf->timezone) > 0)
```

Comment:
Does this account for NULL as opposed to empty string (`""`)? Should we also have a NULL check here?

Reply:
Thanks for the question. bf->timezone can't be NULL. I will add a corresponding comment and an Assert.